### PR TITLE
Fixes 1-Click release for 7.11

### PR DESCRIPTION
**Pull request by kozlovsky**
This PR together with #6545 contains fixes for various problems with Tribler builds.
With this fixes, 1-Click release Jenkins job is able to assemble executables and test them with application tester

### CODE REVIEW

File: build/win/makedist_win.bat [hunk 1]
```diff
@@ -46,7 +46,7 @@ REM Arno: When adding files here, make sure tribler.nsi actually
 REM packs them in the installer .EXE
 
 python3 -m pip install --upgrade --no-deps --force-reinstall pydantic --no-binary pydantic
-python3 -m pip install --upgrade --no-deps --force-reinstall typing_extensions --no-binary typing_extensions
+python3 -m pip install --upgrade --no-deps --force-reinstall typing_extensions==3.10.0.2 --no-binary typing_extensions
```
Comment (kozlovsky) [inline build/win/makedist_win.bat:49]: Without --no-binary PyInstaller does not include this module into the Tribler exe file on Windows
The latest version of typing_extension (4.0.0) is currently not working as well.
After #6534 we can move it into requirements with --no-binary specification

File: src/tribler-core/tribler_core/check_os.py [hunk 2]
```diff
@@ -89,8 +89,14 @@ def should_kill_other_tribler_instances(root_state_dir):
     if current_pid != old_pid and old_pid > 0:
         # If the old process is a zombie, simply kill it and restart Tribler
         old_process = psutil.Process(old_pid)
-        logger.info(f'Old process status: {old_process.status()}')
-        if old_process.status() == psutil.STATUS_ZOMBIE:
```
Comment (kozlovsky) [inline src/tribler-core/tribler_core/check_os.py:93]: Sometimes Tribler crashes on psutil.NoSuchProcess exception when reading `old_process.status()`

File: src/tribler-core/tribler_core/start_core.py [hunk 3]
```diff
@@ -174,12 +174,15 @@ def start_tribler_core(base_path, api_port, api_key, root_state_dir, gui_test_mo
     loop = asyncio.get_event_loop()
     loop.set_exception_handler(CoreExceptionHandler.unhandled_error_observer)
 
-    loop.run_until_complete(core_session(config, components=list(components_gen(config))))
-
-    if trace_logger:
-        trace_logger.close()
-
-    process_checker.remove_lock_file()
```
Comment (kozlovsky) [inline src/tribler-core/tribler_core/start_core.py:182]: We must always remove the lock file on Tribler crash to re-run the application tester reliably, so try/finally block is necessary here.

File: src/tribler-gui/tribler_gui/core_manager.py [hunk 4]
```diff
@@ -64,7 +64,12 @@ def on_core_read_ready(self):
         if b'Traceback' in raw_output:
             self.core_traceback = decoded_output
             self.core_traceback_timestamp = int(round(time.time() * 1000))
-        print(decoded_output.strip())  # noqa: T001
```
Comment (kozlovsky) [inline src/tribler-gui/tribler_gui/core_manager.py:67]: Sometimes Tribler crashes when trying to write data to stdout during the shutdown

File: src/tribler-core/tribler_core/start_core.py (hunk 3 above)
Comment (kozlovsky) [reply src/tribler-core/tribler_core/start_core.py:182]: Our lock file mechanics are broken, as it was built on the assumption that OS does not reuse PIDs before system restart. This is not true, Windows especially reuses process IDs quickly, and that leads to many non-obvious bugs. I will open a separate issue for that.

File: src/run_tribler.py [hunk 5]
```diff
@@ -135,7 +139,10 @@ def init_boot_logger():
             logger.info("Shutting down Tribler")
             if trace_logger:
                 trace_logger.close()
+
             # Flush all the logs to make sure it is written to file before it exits
             for handler in logging.getLogger().handlers:
                 handler.flush()
+
+            SentryReporter.global_strategy = SentryStrategy.SEND_SUPPRESSED
```
Comment (kozlovsky) [inline src/run_tribler.py:147]: Without this Sentry suppression, it sometimes happens that an error takes place during the Tribler shutdown. Then Sentry tries to display the dialog window, but the Qt state is already partially destructed, which leads to mysterious crashes.

File: tribler.spec [hunk 6]
```diff
@@ -93,18 +93,18 @@ hiddenimports = [
...
+    'pyaes',
     'pydantic',
-    'PyQt5.QtTest',
     'pyqtgraph',
     'pyqtgraph.graphicsItems.PlotItem.plotConfigTemplate_pyqt5',
     'pyqtgraph.graphicsItems.ViewBox.axisCtrlTemplate_pyqt5',
     'pyqtgraph.imageview.ImageViewTemplate_pyqt5',
+    'PyQt5.QtTest',
+    'requests',
+    'scrypt', '_scrypt',
```
Comment (devos50) [inline tribler.spec:106]: I believe `pyaes`, `sqlalchemy` and `script` are not required anymore. These were dependencies for `bitcoinlib` (yes, we used to have a Bitcoin wallet in Tribler 😁 ) but we remove that library and we apparently forgot to remove them from `tribler.spec`.

**Review by devos50 — APPROVED** [review]
Great set of changes! 👍

File: tribler.spec (hunk 6 above)
Comment (kozlovsky) [reply tribler.spec:106]: Good to know! I'll remove them in a separate PR after additional testing

**Pull request merged**